### PR TITLE
Fix bug with bad requests retries and attempt to reuse closed SSL connection

### PR DESCRIPTION
**Pull request by svetlyak40wt**
All details on fixed problems are in the https://github.com/fukamachi/dexador/issues/137 issue.

### CODE REVIEW

**Comment by svetlyak40wt** [comment]
@fukamachi I see my test failed on Windows. Also, some tests are skipped on Windows. Are there any problems on this OS? Should my connection pool tests be skipped too?

File: src/backend/usocket.lisp [hunk 1]
```diff
@@ -664,8 +664,12 @@
                    `(restart-case
                         (handler-bind ((error
```
Comment (ajberkley) [inline src/backend/usocket.lisp:665]: (handler-bind (((and error (not http-request-failed)) might be clearer?

**Comment by ajberkley** [comment]
> @fukamachi I see my test failed on Windows. Also, some tests are skipped on Windows. Are there any problems on this OS? Should my connection pool tests be skipped too?

The windows backend does not use connection pooling, I think because winhttp does it internally.  Maybe skip the connection pool tests on windows.

**Comment by ajberkley** [comment]
Also, thank you!

File: src/backend/usocket.lisp (hunk 1 above)
Comment (svetlyak40wt) [reply src/backend/usocket.lisp]: Cool. This is working! I've never used such complex type forms with `handler-bind`.

I've pushed a new commit.

**Comment by ajberkley** [comment]
Thank you!  I did some manual testing to verify that it works well.

There is still a bad interaction with timeouts:
(dex:get "http://httpbin.org/delay/5" :use-connection-pool t :read-timeout 10)
(dex:get "http://httpbin.org/delay/5" :use-connection-pool t :read-timeout 1) ;; will retry

But, this is a pre-existing problem...

**Pull request merged**